api/gateway/apimiddleware/process_request.go

```diff
@@ -77,7 +77,8 @@ func (m *ApiProxyMiddleware) PrepareRequestForProxying(endpoint Endpoint, req *h
 // ProxyRequest proxies the request to grpc-gateway.
 func ProxyRequest(req *http.Request) (*http.Response, ErrorJson) {
 	// We do not use http.DefaultClient because it does not have any timeout.
-	netClient := &http.Client{Timeout: time.Minute * 2}
+	netClient := &http.Client{Timeout: cmd.ApiTimeout.Value}
```

Comment:
The `cmd` package should be used as close to the entry point of the app as possible. Please create a new field in https://github.com/prysmaticlabs/prysm/blob/5983d0a397d33954bf98eb655b5fa7f6f6b2cd9a/api/gateway/apimiddleware/api_middleware.go#L14-L17 and fill it with the flag's value on struct initialization.